test/instrumentation/modules/mongodb.test.js

```diff
@@ -206,6 +216,11 @@ test('ensure run context', async function (t) {
         `span ${s.type}.${s.subtype} "${s.name}" is a child of the transaction`,
       );
     });
+    // Internal calls to `createSpan` are not duplicated
+    t.ok(
+      createSpanSpy.hasBeenCalled(3),
+      `spans created: expected 3, actual ${createSpanSpy.calls.length}`,
```

Comment:
nit: I was a bit confused as to why this is 3 and not 4.  It is because `agent.startSpan('manual').end();` doesn't happen to call `agent._instrumentation.createSpan()`, because of an implementation detail.

This is totally optional. A potential alternative would be to use the existing `transaction.span_count.started` value reported on serialized transactions. Like this: https://gist.github.com/trentm/032e3b560dc8ce4c3d1092253a956890

Then `spyOn` wouldn't be needed in this case. I think `spyOn` is still a useful test utility that could be added here, or saved for a separate commit.

Reply:
Oh, I did not know about `transaction.span_count.started`. If its just a counter and there is no internal logic that tweaks its value then I think its better to use that. I'll apply your suggestion, thanks

Reply:
> Oh, I did not know about `transaction.span_count.started`.

Yah, totally fair. It exists for collecting data on dropped spans when hitting `transactionMaxSpans` limits, so it is a little indirect and lucky to be usable for this.